controllers/directoryservice_controller.go

```diff
@@ -171,14 +178,46 @@ func (r *DirectoryServiceReconciler) Reconcile(ctx context.Context, req ctrl.Req
 	return requeue, nil
 }
 
-// SetupWithManager stuff
+// SetupWithManager function defines which objects the operator owns or watches then sends
+// reconcile requests to the Reconcile method based on those objects.
 func (r *DirectoryServiceReconciler) SetupWithManager(mgr ctrl.Manager) error {
-	// add this line
+
+	// Index the spec.passwords field in the DirectoryService spec and return a list of password objects for the controller manager to watch.
+	if err := mgr.GetFieldIndexer().IndexField(context.Background(), &directoryv1alpha1.DirectoryService{}, passwordField, func(rawObj client.Object) []string {
+        // Extract the Password name from the DirectoryService Spec, if one is provided
+        directoryService := rawObj.(*directoryv1alpha1.DirectoryService)
+        if directoryService.Spec.Passwords == nil {
+            return nil
+        }
+        return []string{
+			directoryService.Spec.Passwords["uid=openam_cts,ou=admins,ou=famrecords,ou=openam-session,ou=tokens"].SecretName,
+			directoryService.Spec.Passwords["uid=am-identity-bind-account,ou=admins,ou=identities"].SecretName,
```

Comment:
this still has hard coded DN key names.

Keep in mind that the operator could be used in non AM / IDM contexts.

let's have a rethink on this objective